docs/adr/adr_database-history.md

```diff
@@ -0,0 +1,150 @@
+# ADR: Database history
+
+[Trello-card](https://trello.com/c/KQpZaHqC)
+
+Decision Deadline: Feb 2023
+
+Discussion Participants: @pylipp, @aerinsol, @haguesto, @vahidbazzaz
+
+## Status
+
+Proposed
+
+## Context or Problem Statement
+
+At the moment, we do not store relevant data about changes in stock, products, etc. (or, at least, in a rather convoluted way). Hence we cannot provide data insights about stock history to our partner which in turn would support them to analyse trends and communicate metrics to stakeholders.
+
+## Decision Drivers
+
+The technological solution for the problem requires
+
+1. Simplicity: Our application is fairly small. We don't have capacities to neither dive into complex frameworks nor to maintain them nor to onboard new developers for them. The format of the recorded data should be clear such that querying and analysing data is straightforward.
+1. Extensibility: it should not be tricky if different kinds of data need to be recorded, or if existing data schemas change
+1. Robustness: we have to make sure that the data recorded represents the actual data at that point in time when it's read later
+1. Security: every change to the DB should be recorded. Recorded data should not be modifiable
+1. Disk space: is cheap and only of minor concern here
+1. Integration in both dropapp and boxtribute 2.0
+
+## Considered Options
+
+There are five options with an impact on different levels of the software.
+
+### Audit table
+
+This is a record of changes, tracking change type, change time, change description, name of the table that the change was made in, ID of the changed record, and possibly a structured payload. The current `history` table implements this concept.
+A variation is to have another table to record details for each change.
+
+*Advantages*
+
+- changes of data are recorded
+- simple and extensible
+- can be combined with database triggers to record any change on DB level
+
+*Drawbacks*
+
+- at a later point in time, the historical state of the data cannot be clearly derived: one can see that a field of a row has changed but other fields might have changed as well
+- dealing with data type conversion when storing/reading change payload ("old"/"new" value)
```

Comment:
also not sure about this. I do not think we have data type conversions in the history table, but it is annoying to store different types in the same table.

Reply:
okay, I should have put it like this: dealing with different data types requires additional effort